content/tools.md

```diff
@@ -121,20 +122,82 @@ gradle cucumber
 
 ### Rake
 
-This requires a `Rakefile` with a `features` task definition. For example:
+Using Rake requires a `Rakefile` with a `features` task definition.  For example:
 
 ```ruby
+require 'rubygems'
+require 'cucumber'
 require 'cucumber/rake/task'
 
 Cucumber::Rake::Task.new(:features) do |t|
   t.cucumber_opts = "--format pretty" # Any valid command line option can go here.
 end
 ```
 
-If you are using [Ruby on Rails](/implementations/ruby/ruby-on-rails/), this task is defined for you already.
+This will run all the Features with the pretty formatter.
+
+Note, how we use the `cucumber_opts` accessor to define our arguments passed to Cucumber.
+
+If you are using [Ruby on Rails](#ruby-on-rails), this task is defined for you already.
 
 Now you can run Cucumber with Rake:
 
 ```shell
 rake features
-```
\ No newline at end of file
+```
+
+#### Using Profiles in Rake Tasks
+
+For complex Feature runs that are tested often, it is nice to save the command line arguments as [Cucumber profiles](/cucumber/configuration/#defining-profiles).
+
+Once you have some profiles defined, you can use them in your Rake tasks, like so:
+
+```ruby
+require 'rubygems'
+require 'cucumber'
+require 'cucumber/rake/task'
+
+namespace :features do
+  Cucumber::Rake::Task.new(:non_js) do |t|
+    t.profile = "webrat"
+  end
+
+  Cucumber::Rake::Task.new(:selenium) do |t|
+    t.profile = "selenium"
+  end
+end
+```
+
+#### Guarding Your Production Boxes From Cucumber
+
+Since Rake tasks are used on development and productions systems, it is generally a good idea to place a guard around your Cucumber task so your productions boxes don't need to install Cucumber.
+
+Below is an example of how to do this.  This example is the Rake task that Cucumber generates for Rails projects, but the same idea applies to any project using Cucumber and Rake:
+
+```ruby
+require 'rubygems'
+
+begin
+  require 'cucumber'
+  require 'cucumber/rake/task'
+
+  Cucumber::Rake::Task.new(:features) do |t|
+    t.cucumber_opts = "--format pretty"
+  end
+
+  task :features => 'db:test:prepare'
+rescue LoadError
```

Comment:
is this spacing right?

Reply:
I don't know (it's Ruby). Please fix as needed?

Reply:
Thnx @olleolleolle :) 
If you don't mind, would you please have a look at the documentation preview on https://deploy-preview-132--cucumber.netlify.com/ and check if the Ruby information is correct/up to date?
In particular the changed files: 
* https://deploy-preview-132--cucumber.netlify.com/cucumber/state/
* https://deploy-preview-132--cucumber.netlify.com/cucumber/configuration/
* https://deploy-preview-132--cucumber.netlify.com/cucumber/continuous-integration/

Reply:
the only thing that stood out was

- Configuration - the config.yml YAML file represented was shown in an unnecessary ordered list which made it harder to read.
- Mentioning Webrat first, then RSpec is the wrong order of popularity

Reply:
I've updated based on these review comments ;)